torch/csrc/api/include/torch/nn/modules/transformercoder.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <torch/nn/cloneable.h>
+#include <torch/nn/module.h>
+#include <torch/nn/modules/container/any.h>
+#include <torch/nn/modules/container/modulelist.h>
+#include <torch/nn/options/transformercoder.h>
+#include <torch/nn/pimpl.h>
+#include <torch/nn/modules/common.h>
+
+#include <torch/types.h>
+
+#include <ostream>
+
+namespace torch {
+namespace nn {
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ TransformerEncoder ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// TransformerEncoder module.
+/// See https://pytorch.org/docs/master/generated/torch.nn.TransformerEncoder.html to
+/// learn abouut the exact behavior of this encoder layer module.
+///
+/// See the documentation for `torch::nn::TransformerEncoder` class to learn what
+/// constructor arguments are supported for this encoder module.
+///
+/// Example:
+/// ```
+/// TransformerEncoderLayer encoderLayer(TransformerEncoderLayerOptions(512, 8).dropout(0.1));
+//  TransformerEncoder encoder(TransformerEncoderOptions(encoderLayer, 6).norm(LayerNorm(LayerNormOptions({2}))));
+/// ```
+class TORCH_API TransformerEncoderImpl : public Cloneable<TransformerEncoderImpl> {
+
+  public:
+    explicit TransformerEncoderImpl(TransformerEncoderOptions options_);
+
+    Tensor forward(
+      const Tensor& src,
+      const Tensor& src_mask = {},
+      const Tensor& src_key_padding_mask = {});
```

Comment:
Is the convention in the C++ API to use undefined tensors instead of `optional<Tensor>`? (Asking for my understanding)

Reply:
Yes. And actually the meaning here is that, this is default to undefined tensor not optional to undefined tensor.